crates/ra_ide/src/goto_definition.rs

```diff
@@ -20,7 +20,7 @@ pub(crate) fn goto_definition(
 ) -> Option<RangeInfo<Vec<NavigationTarget>>> {
     let file = db.parse_or_expand(position.file_id.into())?;
     let original_token =
-        file.token_at_offset(position.offset).filter(|it| !it.kind().is_trivia()).next()?;
+        file.token_at_offset(position.offset).find(|it| it.kind() == SyntaxKind::IDENT)?;
```

Comment:
Hm, does this pass tests? I *think* we have goto def working for tuple fields (`foo.0` syntax), so if tests pass then we miss one!

Reply:
I think ideally this shold work like pick_best in extend selection: https://github.com/rust-analyzer/rust-analyzer/blob/9c9f4635b4eb7d987717b4064a7de789f7983e2a/crates/ra_ide/src/extend_selection.rs#L141

Reply:
yeah, test pass currently. i'll add a test case for that.

Reply:
do you think the solution i used for `hover` i.e. just trying both tokens is acceptable here?

Reply:
i see, pick_best seems like a good solution, i'll try implementing that for the gotos and hover

Reply:
No strong opinion here, but like the `pick_test` solution slightly more. i.e, we chose based only on  the kind of the token (but do this properly, prefering identifiers over numbers over punctuation)